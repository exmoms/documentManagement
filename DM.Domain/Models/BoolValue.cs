namespace DM.Domain.Models
{
    public class BoolValue
    {
        // Forgien-key to metaDataAttribute
        public int MetaDataAttributeId { get; set; }
        public MetaDataAttribute MetaDataAttribute { get; set; }
        // Forgien-key to DocumentVersion
        public int MinDocumentVersionId { get; set; }
        public DocumentVersion DocumentVersion { get; set; }
        public int? MaxDocumentVersionId { get; set; }
        public bool? Value { get; set; }
    }
}
