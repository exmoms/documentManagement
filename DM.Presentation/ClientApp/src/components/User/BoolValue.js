import React, { Component } from "react";
import Checkbox from "@material-ui/core/Checkbox";
import FormControlLabel from "@material-ui/core/FormControlLabel";

class BoolValue extends Component {
  constructor(props) {
    super(props);
    this.state = { value: false };
    this.handleValueChange = this.props.handleValueChange;
  }

  handleChange = (event, checked) => {
    this.setState(
      { value: checked },
      this.handleValueChange(checked, this.props.label)
    );
  };

  render() {
    console.log("check", this.state.value);
    return (
      <div>
        <FormControlLabel
          control={
            <Checkbox
              checked={this.state.value}
              onChange={this.handleChange}
              value="primary"
              inputProps={{ "aria-label": "primary checkbox" }}
            />
          }
          label={this.props.attributeName}
        />
      </div>
    );
  }
}

export default BoolValue;
