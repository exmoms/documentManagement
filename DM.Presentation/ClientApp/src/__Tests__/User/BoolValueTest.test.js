import React from "react";
import ReactDom from "react-dom";
import BoolValue from "../../components/User/BoolValue";
import { cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";
import { act } from "react-dom/test-utils";
import { mount } from "enzyme";
import Checkbox from "@material-ui/core/Checkbox";
import FormControlLabel from "@material-ui/core/FormControlLabel";
afterEach(cleanup);
test("BoolValue matches snapshot", () => {
  const component = renderer.create(<BoolValue />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

describe("BoolValue", () => {
  test("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDom.render(<BoolValue></BoolValue>, div);
  });

  it("BoolValue Functionality ", () => {
    let labelMock = "Text Value";
    let handleValueChangeMock = jest.fn();
    let attributeNameMock = "attrib1";
    let wrapper = mount(
      <BoolValue
        label={labelMock}
        handleValueChange={handleValueChangeMock}
        attributeName={attributeNameMock}
      />
    );
    act(() => {
      let checkbox = wrapper.find(FormControlLabel);
      checkbox
        .prop("control")
        .props.onChange({ target: { checked: true }, labelMock });
    });

    wrapper.update();
    expect(handleValueChangeMock).toHaveBeenCalledTimes(1);
    expect(handleValueChangeMock.mock.calls[0][1]).toStrictEqual(labelMock);
  });
});
