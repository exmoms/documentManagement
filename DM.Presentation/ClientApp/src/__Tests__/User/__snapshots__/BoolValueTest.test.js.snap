// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BoolValue matches snapshot 1`] = `
<div>
  <label
    className="MuiFormControlLabel-root"
  >
    <span
      aria-disabled={false}
      className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-13 MuiCheckbox-root MuiCheckbox-colorSecondary MuiIconButton-colorSecondary"
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={null}
    >
      <span
        className="MuiIconButton-label"
      >
        <input
          aria-label="primary checkbox"
          checked={false}
          className="PrivateSwitchBase-input-16"
          data-indeterminate={false}
          onChange={[Function]}
          type="checkbox"
          value="primary"
        />
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
          />
        </svg>
      </span>
    </span>
    <span
      className="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
    />
  </label>
</div>
`;
